docs/madr/decisions/070-resource-identifier.md

```diff
@@ -0,0 +1,363 @@
+
+##### Internal
+
+|                  | Name                                                                                                                                                                 | Correlated Resources |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------|
+| Listener         | `kuma:dns`<br>`kuma:envoy:admin`<br>`kuma:metrics:prometheus`<br>`_kuma:metrics:opentelemetry:<backendName>`                                                         | –                    |
+| VirtualHost      | listener's name                                                                                                                                                      | –                    |
+| Internal Cluster | `kuma:readiness`<br>`kuma:envoy:admin`<br>`kuma:metrics:hijacker`<br>`_kuma:metrics:opentelemetry:<backendName>`<br>`tracing:`<br>`access_log_sink`<br>`ads_cluster` | –                    |
+
+#### Envoy stats
+
+There are 2 fields to specify alternative name for emitting stats – `stat_prefix`, `alt_stat_name`.
+Field `alt_stat_name` is used only for Cluster, and it's automatically converts all `:` to `_` by Envoy when emitting statistics.
+Field `stat_prefix` is used for HTTPConnectionManager, TCPProxy, KafkaBroker (network filter) and RBAC (network filter).
+Unlike `alt_stat_name` the field `stat_prefix` doesn't have any restrictions from Envoy.
+Kuma sanitizes both fields using `SanitizeMetric` function before setting them on Envoy's resource:
+
+```go
+var illegalChars = regexp.MustCompile(`[^a-zA-Z_\-0-9]`)
+
+// We need to sanitize metrics in order to  not break statsd and prometheus format.
+// StatsD only allow [a-zA-Z_\-0-9.] characters, everything else is removed
+// Extra dots breaks many regexes that converts statsd metric to prometheus one with tags
+func SanitizeMetric(metric string) string {
+	return illegalChars.ReplaceAllString(metric, "_")
+}
+```
+
+I can't find explanation why `alt_stat_name` and `stat_prefix` are treated differently by Envoy. 
+When Envoy exports metrics in Prometheus format resource name is always emitted as label value, it's never part of the metric name, i.e:
+```
+envoy_cluster_upstream_cx_total{envoy_cluster_name="localhost_5000"} 0
+```
+so there is no reason to replace `:` with `_`.
+
+If Envoy tried to support StatsD format then there are still a question, why `stat_prefix` emits metrics with `:` but `alt_stat_name` replaces `:` with `_`.
+Also, StatsD format doesn't seem to be supported out-of-the-box by Envoy and requires configuring `stats_sinks` in the static config.
+
```

Comment:
Whatever the reason for this, with our new format, do we end up without the need for specifying `stat_prefix` and `alt_stat_name` ? If that's the case that's probably something we should mention and decide whether we still set it (in case) or we don't

Reply:
Yes, added a note on that

> As the goal of this MADR is to have unified names in Envoy, Metrics, and API, we want `stat_prefix/alt_stat_name` to always be the same as `name`. This makes the use of `stat_prefix/alt_stat_name` unnecessary and once the migration to new naming is completed Kuma won't set these fields anymore.